gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/privacySettings/js/PrivacySettingsWithPrefs.js

```diff
@@ -0,0 +1,240 @@
+            },
+            "afterRender.bindEdit": {
+                "this": "{that}.dom.editButton",
+                method: "click",
+                args: "{that}.renderDialogForEdit"
+            }
+        },
+        invokers: {
+            getClientData: {
+                funcName: "gpii.oauth2.privacySettingsWithPrefs.getClientData",
+                args: ["{arguments}.0", "{arguments}.1", "{that}.options.selectors.serviceName", "{that}.options.selectors.authDecisionId", "{that}.options.selectors.oauth2ClientId"]
+            },
+            popupDialogForRemoval: {
+                funcName: "gpii.oauth2.privacySettingsWithPrefs.popupDialogForRemoval",
+                args: ["{arguments}.0", "{that}"]
+            },
+            renderDialogForEdit: {
+                funcName: "gpii.oauth2.privacySettingsWithPrefs.renderDialogForEdit",
+                args: ["{arguments}.0", "{that}"]
+            }
+        }
+    });
+
+    gpii.oauth2.privacySettingsWithPrefs.getClientData = function (element, buttonSelector, serviceNameSelector, authDecisionIdSelector, oauth2ClientIdSelector) {
+        element = $(element).closest(buttonSelector);
+        return {
+            serviceName: element.siblings(serviceNameSelector).attr("value"),
+            authDecisionId: element.siblings(authDecisionIdSelector).attr("value"),
+            oauth2ClientId: element.siblings(oauth2ClientIdSelector).attr("value")
+        };
+    };
+
+    gpii.oauth2.privacySettingsWithPrefs.initDialog = function (dialogContainer, dialogForRemovalCss, dialogOptions) {
+        return dialogContainer.dialog($.extend(true, {}, {dialogClass: dialogForRemovalCss}, dialogOptions));
+    };
+
+    gpii.oauth2.privacySettingsWithPrefs.createTooltips = function (that) {
+        var editButtons = that.locate("editButton");
+        var removeButtons = that.locate("removeButton");
+
+        var tooltipOptionsForEdit = $.extend(true, {}, that.options.tooltipOptions, {content: that.options.strings.editLabel});
+        var tooltipOptionsForRemove = $.extend(true, {}, that.options.tooltipOptions, {content: that.options.strings.removeLabel});
+
+        fluid.each(editButtons, function (thisEditButton) {
+            fluid.tooltip(thisEditButton, tooltipOptionsForEdit);
+        });
+        fluid.each(removeButtons, function (thisRemoveButton) {
+            fluid.tooltip(thisRemoveButton, tooltipOptionsForRemove);
+        });
+    };
+
+    gpii.oauth2.privacySettingsWithPrefs.popupDialogForRemoval = function (evt, that) {
+        var clientData = that.getClientData(evt.target, that.options.selectors.removeButton);
+        var dialogForRemoval = that.dialogForRemoval;
+        var dialogContent = fluid.stringTemplate(that.options.strings.removeDecisionContent, {serviceName: clientData.serviceName});
+        dialogForRemoval.find(that.options.selectors.removeDecisionContent).html(dialogContent);
+
+        var dialogOptions = {
```

Comment:
Please refactor all of this inline material into a basic dialog component. This can for now have a single concrete grade OKCancelDialog with the two buttons hardwired and events named clickCancel and clickOK. If you are feeling more ambitious you can turn this into a model-based component with an area initially looking like 
clicks: {
    okButton: 0,
    cancelButton: 0
} (synthesized from the "buttons" configuration) and then forward every "click" callback from the button onto a listener which adds 1 to the relevant click count. All of the external click listeners can then be implemented as modelListeners rather than plain listeners.